docs/spec/lightclient/verification.md

```diff
@@ -0,0 +1,904 @@
+problem statement:
+
+
+#### **[LCV-DIST-TERM]**:
+*Core Verification* either *terminates
+successfully* or it *terminates with failure*.
+
+
+### Design choices
+
+#### **[LCV-DIST-STORE]**:
+*Core Verification* has a local data structure called *LightStore* that
+contains light blocks (that contain a header). For each light block we
+record whether it is verified.
+
+
+#### **[LCV-DIST-PRIMARY]**:
+*Core Verification* has a local variable *primary* that contains the PeerID of a full node.
+
+#### **[LCV-DIST-INIT]**:
+*LightStore* is initialized with a header *trustedHeader* that was correctly 
+generated by the Tendermint consensus. We say *trustedHeader* is verified.
+
+### Temporal Properties
+
+#### **[LCV-DIST-SAFE]**:
+It is always the case that every verified header in *LightStore* was 
+generated by an instance of Tendermint consensus.
+
+#### **[LCV-DIST-LIFE]**:
+From time to time, a new instance of *Core Verification* is called with a
+height *targetHeight*. Each instance must eventually terminate. 
+  - If
+     - the  *primary* is correct, and 
+     - *LightStore* always contains a verified header whose age is less than the
+        trusting period,  
+    then *Core Verification* adds a verified header *hd* with height
+    *targetHeight* to *LightStore* and it **terminates successfully**
+	
+
+> These definitions imply that if the primary is faulty, a header may or
+> may not be added to *LightStore*. In any case,
+> [**[LCV-DIST-SAFE]**](#lcv-vc-inv) must hold.
+
+> The invariant [**[LCV-DIST-SAFE]**](#lcv-dist-safe) and the liveness 
+> requirement [**[LCV-DIST-LIFE]**](#lcv-dist-life)
+> allow that verified headers are added to *LightStore* whose 
+> height was not passed
+> to the verifier (e.g., intermediate headers used in bisection; see below).
+
+> Note that for liveness, initially having a *trustedHeader* within
+> the *trustinPeriod* is not sufficient. However, as this
+> specification will leave some freedom with respect to the strategy
+> in which order to download intermediate headers, we do not give a
+> more precise liveness specification here. After giving the
+> specification of the protocol, we will discuss some liveness
+> scenarios [below](#liveness-scenarios).
+
+### Solving the sequential specification
+
+This specification provides a partial solution to the sequential specification.
+The *Verifier* solves the invariant of the sequential part
+
+[**[LCV-DIST-SAFE]**](#lcv-vc-inv) => [**[LCV-SEQ-SAFE]**](#lcv-seq-inv)
+
+In the case the primary is correct, and there is a recent header in *LightStore*, the verifier satisfies the liveness requirements.
+
+/\ "correct primary"  
+/\ always \E verified header in LightStore. header.Time > now - *trustingPeriod*  
+/\ [**[LCV-A-Comm]**](#lcv-a-comm) /\
+       [**[TMBC-CorrFull]**][TMBC-CorrFull-link] /\
+       [**[LCV-DIST-LIVE]**](#lcv-vc-live)  
+       => [**[LCV-SEQ-LIVE]**](#lcv-seq-live)
```

Comment:
- Uses Unicode to improve readability and avoid mistakenly escaping EOL.
- Adds parens since the markdown does not render the indentations.

```suggestion
⋀ *primary is correct*  
⋀ always ∃ verified header in LightStore. *header.Time* > *now* - *trustingPeriod*  
⋀ [**[LCV-A-Comm]**](#lcv-a-comm) ⋀ (
       ( [**[TMBC-CorrFull]**][TMBC-CorrFull-link] ⋀
         [**[LCV-DIST-LIVE]**](#lcv-vc-live) )
       ⟹ [**[LCV-SEQ-LIVE]**](#lcv-seq-live)
)
```